R/aggregateData.R

```diff
@@ -99,11 +93,12 @@ aggregateData <- function (dat, subunits, units, aggregatemissings = NULL, renam
   # initialize aggregated dataset with subunits to keep
   datAggregated <- dat[ , subunitsToKeep ]
 
-  # check data against standard codes
+  # check aggregatemissings against data
+  am_codes <- unique(c(unlist(dimnames(am))))
   data_codes <- unique(gsub("[[:digit:]]", "vc", unlist(dat[ , subunitsToAggregate])))
 
-  if (any(data_codes %in% standard_codes == FALSE)) {
-    stop("Found nonstandard missing value code(s) in 'dat':", paste(setdiff(data_codes, standard_codes), collapse = ", "), ". Only the following codes are supported:", paste(standard_codes[-length(standard_codes)], collapse = ", "))
+  if (any(data_codes %in% am_codes == FALSE)) {
+    stop("Found the following code(s) in the data but not in 'aggregatemissings': ", paste(setdiff(data_codes, am_codes), collapse = ", "))
```

Comment:
For me, 
`if (any(data_codes %nin% am_codes)) {`
is better readable, but this would require to import `%nin%` from `Hmisc`.